github_test.go

```diff
@@ -119,6 +121,39 @@ var _ = Describe("GitHub Client", func() {
 		})
 	})
 
+	Describe("ListReleases", func() {
+		BeforeEach(func() {
+			source = Source{
+				Owner:      "concourse",
+				Repository: "concourse",
+			}
+		})
+		Context("When list of releases return more then 100 items", func() {
+			BeforeEach(func() {
+				var result []*github.RepositoryRelease
+				for i := 1; i < 102; i++ {
+					result = append(result, &github.RepositoryRelease{ID: github.Int(i)})
+
+				}
+				reqBodyBytes := new(bytes.Buffer)
+				json.NewEncoder(reqBodyBytes).Encode(result)
+
+				server.AppendHandlers(
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", "/repos/concourse/concourse/releases"),
+						ghttp.RespondWith(200, reqBodyBytes.Bytes()),
+					),
+				)
```

Comment:
This won't actually test the pagination feature - in fact, here we're expecting that we only make 1 HTTP request (rather than the 2 we should be getting). That's because we'll have to stub out the pagination headers that Github would normally respond with. Something like the following:

```suggestion
				server.AppendHandlers(
					ghttp.CombineHandlers(
						ghttp.VerifyRequest("GET", "/repos/concourse/concourse/releases", "per_page=100"),
						ghttp.RespondWithJSONEncoded(200, result[:100], http.Header{"Link": []string{`</releases?page=2>; rel="next"`}}),
					),
					ghttp.CombineHandlers(
						ghttp.VerifyRequest("GET", "/repos/concourse/concourse/releases", "per_page=100&page=2"),
						ghttp.RespondWithJSONEncoded(200, result[100:]),
					),
				)
```